### PR TITLE
Make local client Sync

### LINKED ISSUE #814: `std::cell::RefCell<rocket::http::CookieJar>` cannot be shared between threads safely
```
error[E0277]: `std::cell::RefCell<rocket::http::CookieJar>` cannot be shared between threads safely
   --> src/main.rs:359:13
    |
359 |             thread::spawn(move|| {
    |             ^^^^^^^^^^^^^ `std::cell::RefCell<rocket::http::CookieJar>` cannot be shared between threads safely
    |
    = help: within `rocket::local::Client`, the trait `std::marker::Sync` is not implemented for `std::cell::RefCell<rocket::http::CookieJar>`
    = note: required because it appears within the type `std::option::Option<std::cell::RefCell<rocket::http::CookieJar>>`
    = note: required because it appears within the type `rocket::local::Client`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<rocket::local::Client>`
    = note: required because it appears within the type `[closure@src/main.rs:359:27: 364:14 client:std::sync::Arc<rocket::local::Client>, tx:std::sync::mpsc::Sender<rocket::http::Status>]`
    = note: required by `std::thread::spawn`
```

`rocket::local::Client` can't be shared between threads in version 0.4.0-rc.1 while it used can be shared in 0.3.x.

### PR DESCRIPTION
**Pull request by messense**
Fixes #814

### CODE REVIEW

File: core/lib/src/local/request.rs [hunk 1]
```diff
@@ -111,7 +111,8 @@ impl<'c> LocalRequest<'c> {
 
         // Set up any cookies we know about.
         if let Some(ref jar) = client.cookies {
-            for cookie in jar.borrow().iter() {
+            let cookies = jar.read().unwrap();
```
Comment (messense) [inline core/lib/src/local/request.rs]: Not sure how to deal with these `unwrap`s.

**Comment by jebrosen** [comment]
Should we introduce some kind of warning in the documentation that sharing Client between threads can exhibit non-deterministic behavior? There was no risk before cookie tracking and no risk when Client was !Sync, but Sync + cookie tracking opens up the possibility of misuse.

File: core/lib/src/local/request.rs (hunk 1 above)
Comment (messense) [reply core/lib/src/local/request.rs]: Changed to use `expect`

**Comment by SergioBenitez** [comment]
@jebrosen Great idea.

**Comment by SergioBenitez** [comment]
Merged in 64ba2056ef3678152646483c28a20231ad7ae98d. Note added in b16269a30ef9ae9d51878bfd8c6a84622360640d.

**Pull request closed without merging**